desktop/apps/auctions2/templates/index.jsx

```diff
@@ -0,0 +1,35 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { withLayout } from 'desktop/components/react/main_layout'
+
+function Index (props) {
+  const {
+    Layout: { Header, Body, Footer },
+    Templates: { Meta }
+  } = props
+
+  return (
+    <div>
+      <Header>
+        <Meta />
+      </Header>
+      <Body>
+        hello!
+      </Body>
+      <Footer />
+    </div>
```

Comment:
Ahh sorry, let me explain, details are definitely missing :) 

So pug has this Inheritance feature https://pugjs.org/language/inheritance.html that allows you to break apart templates into "zones" and inject that content into "blocks", effectively acting as a portal. 

```
block foo

block foo
  .bar 
    | this content will be injected into the top named block
```
The way force was designed is that the main layout has `block header`, `block `body` and `block footer` scattered about. These are the portals in which sub-apps, when mounted, inject their content into. Each app usually has an `index.jade` that imports `main_layout`, thereby setting it up to receive content imported from a local `header.jade` and `footer.jade` , etc. each split based upon their own tagged `block header` sections. 

Right now this whole process is a black box and _given the architecture that already exists_ there is no way to use anything but jade to get content into those portals. Hence the need to split them up and define a system that renders the content into three _generic_ sections -- header, body, and footer -- which then accepts, as `children` subapp template code. 

I should have used something other than `meta` as an example, but that simply refers to the filename, and could be anything. What it represents is simply a reactified block of jade code.

Some more docs would have helped:

```javascript
res.view('index.jsx', {
  // Anything under `templates` is transformed into a react component 
  // and can be used wherever
  templates: {
     myTemplate: 'my_template.jade' // => <MyTemplate />
  }
})

// templates/index.jsx

function Index (props) {
  const {
    Layout: { Header, Body, Footer },
    Templates: { MyTemplate }
  } = props

  return (
    <div>
      <Header>
        <MyTemplate />
      </Header>
      <Body>
        ... 
      </Body>
      <Footer />
    </div>
  )
}
```

Since it is typical to have jade fragments that go in the header or the footer or the body -- delineated by `block` portals -- those three sections have to be returned and available to accept content. 

Does this make sense? Its really quite simple -- anything passed into `res.view` under the key `templates` is turned into a react component. Use it however. Jade content in, React component out. `<Meta />` corresponds to `meta.jade` in my local app templates folder. Its just an example, commonly seen throughout force. Its not actually a formal `<meta>` html tag, but i can see how that could confuse. I've been in force land and am trying to keep things 1-1 and as feature and complexity-free as possible as i build this out.